packages/langium/src/services.ts

```diff
@@ -35,42 +26,33 @@ import type { ScopeProvider } from './references/scope-provider.js';
 import type { JsonSerializer } from './serializer/json-serializer.js';
 import type { ServiceRegistry } from './service-registry.js';
 import type { AstReflection } from './syntax-tree.js';
+import type { MaybePromise } from './utils/promise-utils.js';
 import type { DocumentValidator } from './validation/document-validator.js';
 import type { ValidationRegistry } from './validation/validation-registry.js';
 import type { AstNodeDescriptionProvider, ReferenceDescriptionProvider } from './workspace/ast-descriptions.js';
 import type { AstNodeLocator } from './workspace/ast-node-locator.js';
 import type { ConfigurationProvider } from './workspace/configuration.js';
 import type { DocumentBuilder } from './workspace/document-builder.js';
-import type { LangiumDocumentFactory, LangiumDocuments } from './workspace/documents.js';
+import type { LangiumDocumentFactory, LangiumDocuments, TextDocumentProvider } from './workspace/documents.js';
 import type { FileSystemProvider } from './workspace/file-system-provider.js';
 import type { IndexManager } from './workspace/index-manager.js';
-import type { WorkspaceManager } from './workspace/workspace-manager.js';
-import type { LanguageServer } from './lsp/language-server.js';
-import type { SignatureHelpProvider } from './lsp/signature-help-provider.js';
-import type { TypeDefinitionProvider } from './lsp/type-provider.js';
-import type { ImplementationProvider } from './lsp/implementation-provider.js';
-import type { CallHierarchyProvider } from './lsp/call-hierarchy-provider.js';
-import type { CodeLensProvider } from './lsp/code-lens-provider.js';
-import type { DeclarationProvider } from './lsp/declaration-provider.js';
-import type { DocumentLinkProvider } from './lsp/document-link-provider.js';
-import type { DocumentUpdateHandler } from './lsp/document-update-handler.js';
-import type { DocumentationProvider } from './documentation/documentation-provider.js';
-import type { FileOperationHandler } from './lsp/file-operation-handler.js';
-import type { InlayHintProvider } from './lsp/inlay-hint-provider.js';
-import type { CommentProvider } from './documentation/comment-provider.js';
-import type { WorkspaceSymbolProvider } from './lsp/workspace-symbol-provider.js';
-import type { NodeKindProvider } from './lsp/node-kind-provider.js';
-import type { FuzzyMatcher } from './lsp/fuzzy-matcher.js';
-import type { IParserErrorMessageProvider } from 'chevrotain';
-import type { TypeHierarchyProvider } from './lsp/type-hierarchy-provider.js';
-import type { AsyncParser } from './parser/async-parser.js';
 import type { WorkspaceLock } from './workspace/workspace-lock.js';
+import type { WorkspaceManager } from './workspace/workspace-manager.js';
+
+export type {
+    InitializeParams, InitializedParams
+};
+
+export interface InitializableService {
+    initialize(params: InitializeParams): void
+    initialized(params: InitializedParams): MaybePromise<void>
+}
```

Comment:
Looking at this again, I'd rather remove this interface: we already have `onInitialize` and `onInitialized` events in the LanguageServer service, which are the recommended ways to hook into the LSP lifecycle. The interface is only used by two core framework classes which require to invert the dependency. The interface is not necessary because LanguageServer hard-codes the calls to these classes anyway.